test/js/test.hopscotch.js

```diff
@@ -155,6 +155,29 @@ describe('Hopscotch', function() {
       expect(hopscotch.getCurrStepNum()).toBe(1);
     });
 
+    it('should reject tour IDs that include invalid characters', function(){
+      try {
+        hopscotch.startTour({
+          id: '(this is a bad tour id!)',
+          steps: [
+            {
+              target: 'shopping-list',
+              orientation: 'left',
+              title: 'Shopping List',
+              content: 'It\'s a shopping list'
+            }
+          ]
+        });
+      }
+      catch (e) {
+        expect(true).toBeTruthy();
+        hopscotch.endTour();
+        return;
+      }
+      expect(false).toBeTruthy();
```

Comment:
Or, according to [this StackOVerflow answer](http://stackoverflow.com/questions/4144686/jasmine-how-to-write-a-test-which-expects-an-error-to-be-thrown),  you could also do:

      expect(function(){
         hopscotch.startTour({
            id: '(this is a bad tour id!)',
            ....
         });
       }).toThrow(new Error("Tour ID is using an invalid format...");